### PR TITLE
Mention the literal macro matcher

### PR DESCRIPTION
**Pull request by da-x**
This feature will soon stabilize in https://github.com/rust-lang/rust/pull/56072.

### CODE REVIEW

**Comment by da-x** [comment]
https://github.com/rust-lang/rust/pull/56072 got merged.

@ehuss I didn't fully understand your comments, but I tried to improve upon the former changes anyway. Perhaps you can pick up from there.

File: src/macros-by-example.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ syntax named by _designator_. Valid designators are:
 * `meta`: a [_MetaItem_], the contents of an attribute
 * `lifetime`: a [LIFETIME_TOKEN]
 * `vis`: a [_Visibility_] qualifier
+* `literal`: a [_LiteralPattern_] token. Examples: `4`, `-3`, `"String"`
```
Comment (ehuss) [inline src/macros-by-example.md]: This doesn't quite capture that all literals can start with `-`.  I would do something like this for now:

```suggestion
* `literal`: matches `-`<sup>?</sup>[_LiteralExpression_]
```

File: src/macros-by-example.md [hunk 2]
```diff
@@ -85,6 +86,7 @@ syntax named by _designator_. Valid designators are:
 [_TypePath_]: paths.html#paths-in-types
 [_Type_]: types.html#type-expressions
 [_Visibility_]: visibility-and-privacy.html
+[_LiteralPattern_]: patterns.html#literal-patterns
```
Comment (ehuss) [inline src/macros-by-example.md]: This should be alphabetized.

**Comment by ehuss** [comment]
Thanks! :smile:

**Pull request merged**